controllers/deployment_controller.go

```diff
@@ -64,6 +70,30 @@ func (r *DeploymentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	}
 	log.Info("Custom resource obtained")
 
+	if deployment.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Instance is not being deleted, add the finalizer if not present
+		if !containsFinalizer(deployment) {
+			deployment.ObjectMeta.Finalizers = append(deployment.ObjectMeta.Finalizers, csiFinalizer)
+			if err = r.Client.Update(ctx, deployment); err != nil {
+				log.Error(err, "Error adding finalizer")
+				return ctrl.Result{Requeue: true}, err
+			}
+
+			log.Info("Successfully add finalizer to CR")
+		}
+	} else {
+		if containsFinalizer(deployment) {
```

Comment:
From my point of view, finalizer is needed to execute some pre-delete logic before resource deletion. I don't see the point of using finalizers here, if we just remove it after deletion and don't do anything. I suppose, this logic works the same as if we don't have finalizer and operator doesn't handle resource deletion. Am I wrong about it?

Reply:
Discussed in a meeting